client/src/js/components/bhPatientGroupSelect.js

```diff
@@ -0,0 +1,43 @@
+angular.module('bhima.components')
+  .component('bhPatientGroupSelect', {
+    templateUrl : 'modules/templates/bhPatientGroupSelect.tmpl.html',
+    controller  : PatientGroupSelectController,
+    transclude  : true,
+    bindings    : {
+      patientGroupUuid : '<',
+      onSelectCallback : '&',
+      label            : '@?',
+      required         : '<?',
+      validateTrigger  : '<?',      
+    },
+  });
+
+PatientGroupSelectController.$inject = [
+  'PatientGroupService'
+];
+
+/**
+ * Patient Group selection component
+ *
+ */
+function PatientGroupSelectController(PatientGroups) {
+  var $ctrl = this;
+
+  $ctrl.$onInit = function onInit() {    
+    // fired when a patient group has been selected
+    $ctrl.onSelectCallback = $ctrl.onSelectCallback || angular.noop;
```

Comment:
Since this is required, you don't actually need to instantiate it - angular will default to `angular.noop` if it is not defined.